app/services/subprocess_runner.py

```
@@ -130,6 +130,23 @@ def verify_timeout(owner: str | None = None) -> int:
     return max(0, int(_resolve_policy("verify_timeout", owner) or 0))
 
 
+def output_size_progress(current: int, expected_size: int) -> int:
+    """Estimate a 5–95% progress value from output-file growth.
```

Ruff (0.15.18)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)